mergify_engine/queue/queue_freeze.py

```diff
@@ -0,0 +1,124 @@
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+import dataclasses
+import datetime
+import typing
+
+import daiquiri
+import msgpack
+
+from mergify_engine import date
+from mergify_engine import utils
+from mergify_engine.context import Repository
+
+
+LOG = daiquiri.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class QueueFreeze:
+
+    repository: Repository
+
+    # Stored in redis
+    name: str = dataclasses.field(metadata={"description": "Queue name"})
+    reason: str = dataclasses.field(metadata={"description": "Freeze message"})
+    user_app_name: str = dataclasses.field(
+        metadata={"description": "Application responsible for freeze"}
+    )
+    freeze_date: datetime.datetime = dataclasses.field(default_factory=date.utcnow)
+    freeze_state: bool = True
+
+    class Serialized(typing.TypedDict):
+        name: str
+        reason: str
+        user_app_name: str
+        freeze_date: datetime.datetime
+        freeze_state: bool
+
+    @classmethod
+    async def get(
+        cls, repository: Repository, queue_name: str
+    ) -> typing.Optional["QueueFreeze"]:
+
+        queue_freeze = cls(
+            repository=repository,
+            name=queue_name,
+            reason="",
+            user_app_name="",
+            freeze_date=datetime.datetime.utcnow(),
+            freeze_state=False,
+        )
+
+        with utils.yaaredis_for_queue() as redis_queue:
+            queue_freeze_data_raw = await redis_queue.hget(
+                queue_freeze._get_redis_hash(),
+                queue_freeze._get_redis_key(),
+            )
+
+            if queue_freeze_data_raw is not None:
+                # NOTE(Syffe): timestamp parameter means that timestamp variables will be converted to
+                # datetime (value 3=to_datetime()). Other values can be used: 1=to_float(), 2=to_unix_ns()
+                queue_freeze_data = msgpack.unpackb(queue_freeze_data_raw, timestamp=3)
+
+                return cls(
+                    repository=repository,
+                    name=queue_freeze_data["name"],
+                    reason=queue_freeze_data["reason"],
+                    user_app_name=queue_freeze_data["user_app_name"],
+                    freeze_date=queue_freeze_data["freeze_date"],
+                    freeze_state=queue_freeze_data["freeze_state"],
+                )
+
+            else:
+                return None
+
+    def _get_redis_hash(self) -> str:
+        return f"merge-freeze~{self.repository.installation.owner_id}"
+
+    def _get_redis_key(self) -> str:
+        return f"{self.repository.repo['id']}~{self.name}"
+
+    def get_json(self) -> typing.Dict[str, typing.Any]:
```

Comment:
```suggestion
    def jsonify(self) -> typing.Dict[str, typing.Any]:
```
I think this is the name we use usually.

Might be worth using a `TypedDict` also?